src/api/app/controllers/assignments_controller.rb

```diff
@@ -0,0 +1,55 @@
+class AssignmentsController < ApplicationController
+  before_action :set_project, only: :index
+  before_action :set_package, only: %i[create destroy]
+  before_action :set_assignee, only: :create
+
+  # GET /assignments/projects/:project_name
+  def index
+    assignments = @project.assignments
+
+    render 'assignments/index', locals: { assignments: assignments, project: @project }, formats: [:xml]
+  end
+
+  # POST /assignments/projects/:project_name/packages/:package_name
+  def create
+    @assignment = Assignment.new(assigner: User.session, assignee: @assignee, package: @package)
+
+    if @assignment.save
+      render_ok
+    else
+      render_error message: @assignment.errors.full_messages.to_sentence,
+                   status: 400, errorcode: 'invalid_assignment'
+    end
+  end
+
+  # DELETE /assignments/projects/:project_name/packages/:package_name
+  def destroy
+    @assignment = @package.assignment
+    @assignment.destroy if @assignment
```

Comment:
Shouldn't this return 404 if the `@assignment` is not found?

Reply:
The DELETE operation in a RESTful API interface should be idempotent. See https://restfulapi.net/idempotent-rest-apis/

Reply:
Very true. :grimacing:

Changed. Thanks!